go/rpc/network_client.go

```diff
@@ -31,10 +33,13 @@ func NewEncNetworkClient(rpcAddress string, viewingKey *ViewingKey) (*EncRPCClie
 
 // NewNetworkClient returns a client that can make RPC calls to an Obscuro node
 func NewNetworkClient(address string) (Client, error) {
-	// TODO - Allow instantiator to choose between websockets and HTTP
-	rpcClient, err := rpc.Dial(ws + address)
+	if !strings.HasPrefix(address, http) && !strings.HasPrefix(address, ws) {
```

Comment:
I added this check because if no protocol is provided, it attempts to default to ipc, so I thought it was best to catch that cases explicitly.